File: java/Recipeak/src/main/java/com/revature/services/HistoryService.java
package com.revature.services;

import java.util.List;

import com.revature.beans.History;
import com.revature.beans.Recipe;
import com.revature.beans.User;

public interface HistoryService {
	public History addHistory(History history);
	public History getHistoryById(int id);
	public History updateHistory(History history);
	public void deleteHistory(History history);
	public List<History> historyByUser(User user);
	public History addHistory(User user, Recipe recipe);
	public List<History> getReviewsByUser(User user);
	public List<History> getReviewsByRecipe(Recipe recipe);
	public History addReview(History history);
}

File: recipeak/src/app/history.service.ts
import { Injectable } from '@angular/core';
import { HttpHeaders, HttpClient } from '@angular/common/http';
import { Observable, pipe, of } from 'rxjs';
import { map } from 'rxjs/operators';
import { User } from './user';
import { History } from './history';

@Injectable({
  providedIn: 'root'
})
export class HistoryService {

  private url = '/Recipeak/';
  constructor(public http: HttpClient) { }

  getHistory(id: number) {
    return this.http.get(`${this.url}user/${id}/history`)
    .pipe(map(resp => resp as History[]));
  }

  getReviewsByRecipe(id: number) {
    return this.http.get(`${this.url}recipe/${id}/reviews`)
    .pipe(map(resp => resp as History[]));
  }

  getReviewsByUser(id: number) {
    console.log(`${this.url}user/${id}/reviews`);
    return this.http.get(`${this.url}user/${id}/reviews`)
    .pipe(map(resp => resp as History[]));
  }

  addReview(history: History) {
    return this.http.post(`${this.url}history/review`, history)
    .pipe(map(resp => resp as History));
  }
}